### PR TITLE
Feature/376/blue

### LINKED ISSUE #376: Loan books
## ✨ Feature request
The user clicks on "Låna bok" button and an input bar pops up where you should write down your books code, after that you press the "Låna" button and you are done.
### 📇 User story
As a user I want to be able to loan a book by scanning the books code.

### 📜 Acceptance Criteria
- [x] investigate existing functionality
- [x] You can click on the "Låna bok" button and a input bar pops up.
- [x] You can write your books code into the input bar and then push  the "Låna" button.
- [ ] The book will now be shown as loaned.
### 💡 Additional context
add css as mockup.

### PR DESCRIPTION
**Pull request by te4-andre-skvarc**
## ⚙️ What issue did you implement or fix?
Fixes #376 

## 📜 Summary
- Did almost a complete rebuild of the AddLoan component.  Because the old one was broke and didnt work...
- Added comments in hard to understand areas. 
- It works with backend communication if the user is logged in, the book data is saved in the database.  

## 📝 Type of change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## 💩 How can this implementation be improved?
- Add style to the component
- Create a test for the component so it's not broken again.
"was about to write a test but it is hard when you have to be signed in"

### CODE REVIEW

**Review by Te4-Jacob-sandstrom — CHANGES_REQUESTED** [review]
FIx issues with integration tests

File: frontend/src/components/AddLoan.vue [hunk 1]
```diff
@@ -1,90 +1,109 @@
...
+  public menu: boolean = false;
   public inputReturn: string = '';
-  public popoverShow: boolean = false;
 
-  public onClose(): void {
-    this.popoverShow = false;
+  public form: AddLoan = {
+    book_id: 0, //eslint-disable-line camelcase
+    /* Hardcoded value atm for the school */
+    lent_by_id: 2054282603, //eslint-disable-line camelcase
```
Comment (te4-jonathan-ronsten) [inline frontend/src/components/AddLoan.vue:79]: Do we want to have it hardcoded or is their any way to get this in another way?

**Review by te4-jonathan-ronsten — CHANGES_REQUESTED** [review]
Looks good overall, seems to be some failure with lighthouse. See if there is some way to fix this before merging

File: frontend/src/components/AddLoan.vue (hunk 1 above)
Comment (te4-adrian-almetun-smeds) [reply frontend/src/components/AddLoan.vue:79]: This has to be hardcoded somewhere. Preferably it should be hardcoded in the backend but I'd say this works for now. But we should open an issue to implement this in the backend.

**Pull request closed without merging**